src/components/header/Header.jsx

```diff
@@ -1,45 +1,52 @@
-import React from 'react'
+import React, { useContext } from 'react'
 import { Navbar } from 'reactstrap'
 import Icon from 'components/icons/Icon'
+import GlobalContext from 'store/configureContext'
+import { downloadPDF } from 'components/pdf-downloader/PDFDownloader'
 
 import 'components/header/Header.scss'
 
-class Header extends React.Component {
-  componentDidMount() {}
+const Header = () => {
+  const {
+    state: { pdfDocument, pdfUrl },
+  } = useContext(GlobalContext)
```

Comment:
Can you find a way to have a single component, connected to the context? Do you really need the context?

Reply:
Yes, I do. If I connect it to just one component it doesn't make sense to use it at all.

Reply:
I had some thought about using native event to pass basic information. It is more Web-like, so it's more preferable because it's more standard.

I make a piece of code for it but have-not pushed to the repository because I did not have time to finalize it. However, it works and you may be interested in the following snippet:

```javascript
const ToolbarButton = ({ children, action, ...restProps }) => (
  <Button type="button" value={action} className="toolbar-btn" {...restProps}>
    {children}
  </Button>
)

const ToolbarButtonGroup = ({ children, ...restProps }) => (
  <ButtonGroup {...restProps}>{children}</ButtonGroup>
)

const Toolbar = ({ tag: Tag = 'div', onClick, ...restProps }) => (
  <Tag className="toolbar btn-toolbar" {...restProps}>
    <ToolbarButtonGroup>
      <ToolbarButton action="open-outlines" onClick={onClick}>
        Open sidebar with the outlines
      </ToolbarButton>

      <ToolbarButton action="open-thumbnails" onClick={onClick}>
        Open sidebar with the page thumbnails
      </ToolbarButton>
    </ToolbarButtonGroup>

    <ToolbarButtonGroup>
      <ToolbarButton action="info" onClick={onClick}>
        Open information
      </ToolbarButton>
    </ToolbarButtonGroup>

    <ToolbarButtonGroup>
      <ToolbarButton action="download" onClick={onClick}>
        Download
      </ToolbarButton>
      <ToolbarButton action="print" onClick={onClick}>
        Print
      </ToolbarButton>
    </ToolbarButtonGroup>
  </Tag>
)

const Header = ({ ...restProps }) => (
  <Navbar light color="light" className="header" {...restProps}>
    <Toolbar onClick={event => console.log(event.target.value)} />
  </Navbar>
)
```